cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java

```diff
@@ -34,23 +35,34 @@
    * Schedule a program to be run in a defined schedule.
    *
    * @param program Program that needs to be run.
-   * @param programType type of program.
-   * @param schedule Schedule with which the program runs.
+   * @param scheduleSpec specification of the schedule.
    * @throws SchedulerException on unforeseen error.
    */
-  void schedule(Id.Program program, SchedulableProgramType programType, Schedule schedule)
+  void schedule(Id.Program program, ScheduleSpecification scheduleSpec)
     throws SchedulerException;
 
   /**
    * Schedule a program to be run in a defined schedule.
    *
    * @param program Program that needs to be run.
-   * @param programType type of program.
-   * @param schedule Schedule with which the program runs.
-   * @param properties system properties to be passed to the schedule
+   * @param scheduleSpec specification of the schedule.
+   * @param updateMds indicates whether to update the application specification in the MDS with this schedule. This is
+   *                  needed because during application deployment, the app spec in the MDS already contains a schedule
+   *                  spec, so we do not want to add it again/override it.
    * @throws SchedulerException on unforeseen error.
    */
-  void schedule(Id.Program program, SchedulableProgramType programType, Schedule schedule,
+  void schedule(Id.Program program, ScheduleSpecification scheduleSpec, boolean updateMds)
+    throws SchedulerException;
+
+  /**
+   * Schedule a program to be run in a defined schedule.
+   *
+   * @param program Program that needs to be run.
+   * @param scheduleSpec specification of the schedule.
+   * @param properties system properties to be passed to the schedule.
```

Comment:
`scheduleSpec` contains `properties`, and we pass in additional `properties` to `schedule` method. It is a bit confusing as to whether they are the same properties or not.